### PR TITLE
gh-88226: Emit TARGET labels in Python/ceval.c when debugging, even if computed gotos aren't enabled

### PR DESCRIPTION
**Pull request by smontanaro**
This is an updated PR from the original one I submitted (in the before GitHub issues days).

Note: I am still acquainting myself with my new MacBook, so don't have quite the full suite of autoconf tools available. I was thus unable to generate configure from configure.ac. I manually editing configure instead.

* Issue: gh-88226

### CODE REVIEW

**Comment by smontanaro** [comment]
I messed up the reference to the issue number when creating the NEWS blurb. I entered "25949" instead of "88226" (I think that's the right number). I don't know if the generated blurb can simply be `git mv ...`'d or not.

**Comment by smontanaro** [comment]
I think I fixed this problem. I `git rm`'d the first blurb then ran `blurb` again.

**Comment by smontanaro** [comment]
One other thing. I'm confident this does the right thing. I ran these steps to verify that no warnings were issued for unused labels when compiling `Python/ceval.o`:

```
#!/bin/bash

./configure --without-pydebug --with-computed-gotos 2>&1 | egrep -v '^checking'
echo "Computed goto, Py_DEBUG disabled"
rm -f Python/ceval.o
make Python/ceval.o

./configure --with-pydebug --with-computed-gotos 2>&1 | egrep -v '^checking'
echo "Computed goto, Py_DEBUG enabled"
rm -f Python/ceval.o
make Python/ceval.o

./configure --with-pydebug --without-computed-gotos 2>&1 | egrep -v '^checking'
echo "Computed goto disabled, Py_DEBUG enabled"
rm -f Python/ceval.o
make Python/ceval.o

./configure --without-pydebug --without-computed-gotos 2>&1 | egrep -v '^checking'
echo "Computed goto disabled, Py_DEBUG disabled"
rm -f Python/ceval.o
make Python/ceval.o
```

**Comment by smontanaro** [comment]
Clang seems happy with the change as well.

**Comment by smontanaro** [comment]
This still spits out warnings when compiling Python/ceval.c on Windows. I haven't the slightest idea how to control for that, but I assume it must be possible. If you know how this is done, can you make a suggestion?

**Comment by serhiy-storchaka** [comment]
Disabling the warnings globally looks not good to me. It can lead to preserving unused labels in other parts of the code. Why not disable them locally, using compiler-specific pragmas? This is how we handle other types of warnings.

**Comment by smontanaro** [comment]
Update... Thanks to @zooba and @mattip, the latest variation performs a more surgical suppression of the unused label warning. `configure` is no longer affected, and the `TARGET_*` labels are always defined, not just when `Py_DEBUG` is enabled.

File: Python/ceval.c [hunk 1]
```diff
@@ -1021,6 +1021,18 @@ typedef struct {
...
     (call_shape.kwnames == NULL ? 0 : ((int)PyTuple_GET_SIZE(call_shape.kwnames)))
 
+/* Disable unused label warnings.  They are handy for debugging, even
+   if computed gotos aren't used. */
+
+/* TBD - what about other compilers? */
+#ifdef __GNUC__
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wunused-label"
+#else /* MS_WINDOWS */
```
Comment (eryksun) [inline Python/ceval.c]:
```suggestion
#if defined(__GNUC__)
#  pragma GCC diagnostic push
#  pragma GCC diagnostic ignored "-Wunused-label"
#elif defined(_MSC_VER) /* MS_WINDOWS */
```

File: Python/ceval.c [hunk 2]
```diff
@@ -1374,6 +1386,11 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
     goto error;
 
 }
+#ifdef __GNUC__
+#  pragma GCC diagnostic pop
+#else /* MS_WINDOWS */
```
Comment (eryksun) [inline Python/ceval.c]:
```suggestion
#if defined(__GNUC__)
#  pragma GCC diagnostic pop
#elif defined(_MSC_VER) /* MS_WINDOWS */
```

File: configure.ac [hunk 3]
```diff
@@ -2118,7 +2118,7 @@ then
 	case $ac_cv_prog_cc_g in
 	yes)
 	    if test "$Py_DEBUG" = 'true' ; then
-		OPT="-g $PYDEBUG_CFLAGS -Wall"
+		OPT="-g $PYDEBUG_CFLAGS -Wall -Wno-unused-label"
```
Comment (serhiy-storchaka) [inline configure.ac]: Do we need this?

**Review by serhiy-storchaka — APPROVED** [review]

File: configure.ac (hunk 3 above)
Comment (smontanaro) [reply configure.ac]: No. I'm a Git idiot. It snuck back in somehow.

**Comment by serhiy-storchaka** [comment]
@smontanaro Aren't you a coredev? Don't you have the ability to do that yourself? I just don't promise that I will do it quickly, because we have regular blackouts here (5 minutes until the next blackout). But I will do it in 2 hours or tomorrow.

**Comment by smontanaro** [comment]
@serhiy-storchaka No worries and no rush, just as long as it's on your radar screen. Whenever you have a chance. I gave up my commit privilege a few years ago (don't really want it back).

I wasn't aware you are in Ukraine. Stay well.

**Comment by serhiy-storchaka** [comment]
Done. Don't forget to remove the branch in your GitHub repository and in your local copy if you no longer need it.

**Pull request merged**